PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = ["hr.attendance"]
-    _inherit = ['hr.leave']
-    _inherit = ["hr.leave.accrual.level"]
-    _inherit = ['hr.leave.allocation']
-    _inherit = ['hr.leave.type']
-    _inherit = ['res.users']

Total Changes: 12

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_holidays_attendance/models/hr_attendance.py
+++ b/addons/hr_holidays_attendance/models/hr_attendance.py
@@ -6,7 +6,7 @@ from odoo.osv.expression import AND
 
 
 class HrAttendance(models.Model):
-    _inherit = ["hr.attendance"]
+    _inherit = "hr.attendance"
 
     def _get_overtime_leave_domain(self):
         domain = super()._get_overtime_leave_domain()

--- a/addons/hr_holidays_attendance/models/hr_leave.py
+++ b/addons/hr_holidays_attendance/models/hr_leave.py
@@ -10,7 +10,7 @@ from odoo.tools import float_round
 
 
 class HrLeave(models.Model):
-    _inherit = ['hr.leave']
+    _inherit = 'hr.leave'
 
     overtime_id = fields.Many2one('hr.attendance.overtime', string='Extra Hours')
     employee_overtime = fields.Float(related='employee_id.total_overtime', groups='base.group_user')

--- a/addons/hr_holidays_attendance/models/hr_leave_accrual_plan_level.py
+++ b/addons/hr_holidays_attendance/models/hr_leave_accrual_plan_level.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class HrLeaveAccrualLevel(models.Model):
-    _inherit = ["hr.leave.accrual.level"]
+    _inherit = "hr.leave.accrual.level"
 
     frequency_hourly_source = fields.Selection(
         selection=[

--- a/addons/hr_holidays_attendance/models/hr_leave_allocation.py
+++ b/addons/hr_holidays_attendance/models/hr_leave_allocation.py
@@ -9,7 +9,7 @@ from odoo.osv import expression
 
 
 class HrLeaveAllocation(models.Model):
-    _inherit = ['hr.leave.allocation']
+    _inherit = 'hr.leave.allocation'
 
     def default_get(self, fields):
         res = super().default_get(fields)

--- a/addons/hr_holidays_attendance/models/hr_leave_type.py
+++ b/addons/hr_holidays_attendance/models/hr_leave_type.py
@@ -6,7 +6,7 @@ from odoo import _, api, fields, models
 
 
 class HrLeaveType(models.Model):
-    _inherit = ['hr.leave.type']
+    _inherit = 'hr.leave.type'
 
     overtime_deductible = fields.Boolean(
         "Deduct Extra Hours", default=False,

--- a/addons/hr_holidays_attendance/models/res_users.py
+++ b/addons/hr_holidays_attendance/models/res_users.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     request_overtime = fields.Boolean(compute='_compute_request_overtime')
 
